otherlibs/win32unix/realpath.c

```diff
@@ -0,0 +1,78 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*                         The OCaml programmers                          */
+/*                                                                        */
+/*   Copyright 2020 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#define CAML_INTERNALS
+
+/*
+ * Windows Vista functions enabled
+ */
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x0600
+
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/alloc.h>
+#include <caml/fail.h>
+#include <caml/osdeps.h>
+#include "unixsupport.h"
+
+#include <windows.h>
+#include <stdio.h>
+
+CAMLprim value unix_realpath (value p)
+{
+  CAMLparam1 (p);
+  HANDLE h;
+  wchar_t *wp;
+  wchar_t *wr;
+  DWORD wr_len;
+  value rp;
+
+  caml_unix_check_path (p, "realpath");
+  wp = caml_stat_strdup_to_utf16 (String_val (p));
+  h = CreateFile (wp, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING,
+                  FILE_FLAG_BACKUP_SEMANTICS, NULL);
```

Comment:
```suggestion
  h = CreateFile (wp, 0, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, NULL, OPEN_EXISTING,
                  FILE_FLAG_BACKUP_SEMANTICS, NULL);
```
`0` allows access to metadata only, the share constants indicate what you're happy to have happen to the file while it's open.

Reply:
Done in 7ca6ecf